File: src/tests/system/t009.c
/*
 * PyMite - A flyweight Python interpreter for 8-bit and larger microcontrollers.
 * Copyright 2002 Dean Hall.  All rights reserved.
 * PyMite is offered through one of two licenses: commercial or open-source.
 * See the LICENSE file at the root of this package for licensing details.
 */

/**
 * Regression Test for Issue #9
 * Fix BINARY_SUBSCR for case stringobj[intobj]
 */

#include "pm.h"


extern unsigned char usrlib_img[];


int main(void)
{
    PmReturn_t retval;

    retval = pm_init(MEMSPACE_PROG, usrlib_img);
    PM_RETURN_IF_ERROR(retval);

    retval = pm_run((uint8_t *)"t009");
    return (int)retval;
}

File: src/tests/system/t009.py
#
# PyMite - A flyweight Python interpreter for 8-bit and larger microcontrollers.
# Copyright 2002 Dean Hall.  All rights reserved.
# PyMite is offered through one of two licenses: commercial or open-source.
# See the LICENSE file at the root of this package for licensing details.
#

#
# Regression Test for Issue #9
# Fix BINARY_SUBSCR for case stringobj[intobj]
#

import sys

s = "test"
if s[0] != 't':
    sys.exit(1)
